lib/appInsightsWrapper.js

```diff
@@ -0,0 +1,171 @@
+'use strict';
+
+var crypto = require('crypto');
+var appInsight = require('applicationinsights');
+var settings = require('./settings.js');
+var os = require('os');
+var promise = require('bluebird');
+var getos = promise.promisify(require('getos'));
+var getmacs = promise.promisify(require('macaddress').all);
+
+var propertyDict = {};
+var _isStarted = false;
+
+function initPkgInfo() {
+  if (!propertyDict.currentPkgVersion) {
+    require('pkginfo')(module, 'version');
+    propertyDict.currentPkgVersion = module.exports.version || '';
+  }
+  if (!propertyDict.parentPkgName || !propertyDict.parentPkgVersion) {
+    if (module.parent && module.parent.parent) {
+      require('pkginfo')(module.parent.parent, 'name', 'version');
+      propertyDict.parentPkgName = module.parent.parent.exports.name || '';
+      propertyDict.parentPkgVersion = module.parent.parent.exports.version || '';
+    } else {
+      propertyDict.parentPkgName = '';
+      propertyDict.parentPkgVersion = '';
+    }
+  }
+}
+
+function initSubscriptionAndTenantId() {
+  if (propertyDict.subscriptionId && propertyDict.tenantId) {
+    return;
+  }
+  var profile = settings.readProfile();
+  if (profile.subscriptions && profile.subscriptions.constructor === Array && profile.subscriptions.length > 0) {
+    var subs = profile.subscriptions;
+    for (var i = 0; i < subs.length; i++) {
+      if (subs[i].hasOwnProperty('isDefault') && subs[i].isDefault === true) {
+        propertyDict.subscriptionId = subs[i].id;
+        propertyDict.tenantId = subs[i].tenantId;
+        return;
+      }
+    }
+  }
+  // No profile found
+  propertyDict.subscriptionId = '';
+  propertyDict.tenantId = '';
+}
+
+function hasValidMacProperty(obj, key) {
+  return obj &&
+    obj.hasOwnProperty(key) &&
+    obj[key].hasOwnProperty('mac') &&
+    obj[key].mac != '00:00:00:00:00:00';
+}
+
+function initMac(allMacs) {

```

Comment:
rename to 'setMac'?

For this function doesn't do initialisation, and it's different from the 'init' of 'initMacAndTrackEvent'